drivers/gpio/gpio_nrfx.c

```diff
@@ -390,7 +402,8 @@ static const struct gpio_driver_api gpio_nrfx_drv_api_funcs = {
 	.port_clear_bits_raw = gpio_nrfx_port_clear_bits_raw,
 	.port_toggle_bits = gpio_nrfx_port_toggle_bits,
 	.pin_interrupt_configure = gpio_nrfx_pin_interrupt_configure,
-	.manage_callback = gpio_nrfx_manage_callback,
+	.manage_callback = IS_ENABLED(CONFIG_GPIO_NRFX_INTERRUPT) ?
+			   gpio_nrfx_manage_callback : NULL,
```

Comment:
Wont this lead to compilation warnings about `gpio_nrfx_manage_callback()` being unused if `CONFIG_GPIO_NRFX_INTERRUPT=n`?

Reply:
We use `IS_ENABLED`, not `ifdefs` here so that the unused compilation warnings should not happen here (I did not see them locally too)